controllers/capabilities/product_controller.go

```diff
@@ -76,7 +79,37 @@ func (r *ProductReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 	// Ignore deleted Products, this can happen when foregroundDeletion is enabled
 	// https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#foreground-cascading-deletion
-	if product.DeletionTimestamp != nil {
+	if product.GetDeletionTimestamp() != nil && controllerutil.ContainsFinalizer(product, productFinalizer) {
+		controllerutil.RemoveFinalizer(product, productFinalizer)
```

Comment:
To be consistent with the backend controller, the finalizer should be removed after removing the product in 3scale

Reply:
done, thanks